### PR TITLE
Prevent focus on AutocompleteField prior to initializing  jQuery UI autocomplete (due to bug in "change" event firing incorrectly).

### PR DESCRIPTION
**Pull request by patricknelson**
Proposed fix for issue #45. Note that this seems like a hack, but it works well given the root cause of the error and has the lowest impact.

**Also note:** I branched off of the `3.5.2` tag since this affects both `3.5.2` as well as `master` code. I'd suggest you also create create a `3.5` branch (which would presumably be at `3.5.2`) that you can merge this into and then merge into `master`. I did that since there's no `3` branch and `3.4` is outdated.

Thankfully this compatible with both `3.5` and `4` versions of this module, since not much as changed on the JS side of things 😄

### CODE REVIEW

File: javascript/AutocompleteField.js [hunk 1]
```diff
@@ -50,68 +50,73 @@
...
+
+                // Prevent this field from loading itself multiple times
+                if ($input.attr('data-loaded') == 'true')
+                    return;
+                $input.attr('data-loaded', 'true');
+
+                // Prevent this field from initializing with autocomplete while focused. Required to prevent an
+                // accidental misfire in the jQuery UI autocomplete "change" event that occurs if: 1.) Field starts out
+                // with focus and 2.) You click away to remove focus.
+                $input.blur();
```
Comment (patricknelson) [inline javascript/AutocompleteField.js]: Here is the actual change, just a single line of actual code ☝️

**Comment by patricknelson** [comment]
@tractorcow  I noticed in the diff that my comment had different indentation. I twas because my comment was 100% spaces for indentation while most of the other lines were actually an odd mix of both tabs + spaces. Sorry for the ugly commit... but better to address earlier rather than later !

**Comment by patricknelson** [comment]
p.s. For reference, [my branch `3.5-beta`](https://github.com/patricknelson/silverstripe-autocomplete/tree/3.5-beta) includes fixes for both currently outstanding PR's:

- #47
- #48 

Combined and applied starting from tag `3.5.2`.

**Comment by tractorcow** [comment]
I've also cherry picked this into 3.5 branch.

**Comment by tractorcow** [comment]
Oh I never tagged 4.0.0 stable. Tagged both that and 3.5.3.

**Comment by patricknelson** [comment]
Cool, thanks @tractorcow!

**Pull request merged**